torch/_decomp/decompositions.py

```diff
@@ -4025,6 +4025,24 @@ def trunc(self: Tensor, **kwargs) -> Tensor:
     return torch.where(self > 0, torch.floor(self), torch.ceil(self))
 
 
+@register_decomposition([aten.div.Tensor_mode])
+def div_tensor_mode(self, other, *, rounding_mode: Optional[str]) -> Tensor:
+    compute_dtype, result_dtype = utils.elementwise_dtypes(
+        self, other, type_promotion_kind=utils.ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT
+    )
+
+    if rounding_mode is None:
+        return torch.div(self, other)
+    elif rounding_mode == "trunc":
+        return torch.trunc(torch.div(self, other)).to(result_dtype)
```

Comment:
We cannot use true-divide to implement truncation division for integer dtypes as this will introduce rounding errors.

Reply:
AFAIK `div(mode="trunc")` is the only way to get integer truncation division in the pytorch API. It also maps directly only a hardware instruction, so decomposing it doesn't really make sense.

Decomposing float into true-divide + trunc is legitimate though,

Reply:
ah, understood. I see what you mean. We may want to reconsider decomposing this in the first place...